plugins/common/tls/config.go

```diff
@@ -3,13 +3,17 @@ package tls
 import (
 	"crypto/tls"
 	"crypto/x509"
+	_ "embed"
 	"fmt"
 	"os"
 	"strings"
 
 	"github.com/influxdata/telegraf/internal/choice"
 )
 
+//go:embed client.conf
+var clientConfig string
```

Comment:
Hmm not sure I like having `ClientConfig` and `clientConfig`, how about using sampleConfig here as well like we do in other plugins?

Reply:
Yeah that's confusing I agree. Simply `sampleConfig` is probably not enough as that file also contains a `ServerConfig` with different fields... How about `sampleClientConfig`?

Reply:
sounds good to me!

Reply:
What about splitting this code also into a `client.go` and `server.go`?